modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/ClusterWideStorageProfileValidator.java

```diff
@@ -53,33 +51,38 @@ public CompletableFuture<Void> validate(Collection<String> storageProfiles) {
         );
 
         if (missedStorageProfileNames.isEmpty()) {
-            return;
+            return CompletableFuture.completedFuture(null);
         }
 
-        try {
-            missedStorageProfileNames = logicalTopologyService.logicalTopologyOnLeader()
+        return logicalTopologyService.logicalTopologyOnLeader()
                     .thenApply(topologySnapshot -> findStorageProfileNotPresentedInLogicalTopologySnapshot(
                             storageProfiles,
                             topologySnapshot
-                    )).get(10, TimeUnit.SECONDS);
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            String msg = format(
-                    "Storage profile {} doesn't exist in local topology snapshot with profiles [{}], and distributed refresh failed.",
-                    missedStorageProfileNames,
-                    localLogicalTopologySnapshot.nodes().stream().map(LogicalNode::storageProfiles).collect(Collectors.toSet())
-            );
-
-            throw new SqlException(STMT_VALIDATION_ERR, msg, e);
-        }
-
-        if (!missedStorageProfileNames.isEmpty()) {
-            throw new SqlException(STMT_VALIDATION_ERR, format(
-                    "Some storage profiles don't exist [missedProfileNames={}].",
-                    missedStorageProfileNames
-            ));
-        }
-
-        return completedFuture(null);
+                    )).handle((missedProfileNames, e) -> {
+                        if (e != null) {
+                            String msg = format(
+                                    "Storage profiles {} doesn't exist in local topology snapshot with profiles [{}], "
+                                            + "and distributed refresh failed.",
+                                    missedStorageProfileNames,
+                                    localLogicalTopologySnapshot
+                                            .nodes()
+                                            .stream()
+                                            .map(LogicalNode::storageProfiles)
+                                            .collect(Collectors.toSet())
```

Comment:
Let's import `Collectors#toSet()` statically so the code reads `collect(toSet())`, almost in plain English